xbmc/cores/VideoPlayer/DVDInputStreams/DVDFactoryInputStream.cpp

```diff
@@ -37,12 +38,31 @@
 #include "URL.h"
 #include "filesystem/File.h"
 #include "utils/URIUtils.h"
+#include "addons/AddonManager.h"
+#include "addons/InputStream.h"
 
 
 CDVDInputStream* CDVDFactoryInputStream::CreateInputStream(IVideoPlayer* pPlayer, CFileItem fileitem)
 {
   std::string file = fileitem.GetPath();
 
+  ADDON::VECADDONS addons;
+  ADDON::CAddonMgr::GetInstance().GetAddons(addons, ADDON::ADDON_INPUTSTREAM);
+  for (size_t i=0; i<addons.size(); ++i)
+  {
+    std::shared_ptr<ADDON::CInputStream> input(std::static_pointer_cast<ADDON::CInputStream>(addons[i]));
+    ADDON::CInputStream* clone = new ADDON::CInputStream(*input);
+    ADDON_STATUS status = clone->Supports(fileitem) ? clone->Create() : ADDON_STATUS_PERMANENT_FAILURE;
+    if (status == ADDON_STATUS_OK)
+    {
+      if (clone->Supports(fileitem))
+      {
+        return new CInputStreamAddon(fileitem, clone);
+      }
+      delete clone;
```

Comment:
The delete clone should be one more scope out, so it doesn't leak when when status !=ADDON_STATUS_OK.

(coverity 1355259)